src/decode-ipv4.c

```diff
@@ -527,6 +527,9 @@ int DecodeIPV4(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p,
         CLEAR_IPV4_PACKET((p));
         return TM_ECODE_FAILED;
     }
+    if (!PacketIncreaseCheckLayers(p)) {
+        return TM_ECODE_FAILED;
```

Comment:
call before `DecodeIPV4Packet`? If not at least call `CLEAR_IPV4_PACKET`?

Reply:
indeed before